azure/services/roleassignments/roleassignments.go

```diff
@@ -141,10 +141,20 @@ func (s *Service) getVMSSPrincipalID(ctx context.Context) (*string, error) {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "roleassignments.Service.getVMPrincipalID")
 	defer done()
 	log.V(2).Info("fetching principal ID for VMSS")
-	resultVMSS, err := s.virtualMachineScaleSetClient.Get(ctx, s.Scope.ResourceGroup(), s.Scope.Name())
+	spec := &scalesets.ScaleSetSpec{
+		Name:          s.Scope.Name(),
+		ResourceGroup: s.Scope.ResourceGroup(),
+	}
+
+	resultVMSSIface, err := s.virtualMachineScaleSetGetter.Get(ctx, spec)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get principal ID for VMSS")
 	}
```

Comment:
Should this be updated to below?
```suggestion
	if err != nil {
		return nil, errors.Wrap(err, "failed to get resultVMSSIface")
	}
```

Reply:
On second thought, I think we want to leave it as is. Before, we were giving the error when we failed to get the VM, which we're still doing here.